data/finset.lean

```diff
@@ -992,6 +989,57 @@ sort_eq _ _
 
 @[simp] theorem sort_to_finset [decidable_eq α] (s : finset α) : (sort r s).to_finset = s :=
 list.to_finset_eq (sort_nodup r s) ▸ eq_of_veq (sort_eq r s)
+
 end sort
 
+variable [decidable_eq α]
+
+theorem disjoint_iff_inter_eq_empty [decidable_eq α] {s t : finset α} : disjoint s t ↔ s ∩ t = ∅ :=
```

Comment:
Can this `[decidable_eq α]` be removed?

Reply:
I'm not so sure about unifying lattice `disjoint` with finset `disjoint` like this. For one thing, finset disjoint does not require `decidable_eq`, but lattice disjoint applied to finset does, because the lattice instance requires `decidable_eq` (because the intersection operator requires it). This theorem requires `decidable_eq` in any case because of the intersection on the RHS.

Reply:
@digama0 In case it wasn't clear, I was asking because there is a `variable [decidable_eq α]` above. But you're right about the other things.

Reply:
> The idea behind the unguarded variable statement there is that this is the part of the file where we require `decidable_eq` for everything. The stuff which doesn't need it is above that line.

@digama0 Is this a response to https://github.com/leanprover/mathlib/pull/120#discussion_r184876205? If not, I don't understand your comment.

If so, then yes, I understand how it works, but it's very easy to miss it and add something else below for which you don't want the variable. With conditions such as `[decidable_eq α]`, it can be annoying to find out later that it is imposed even though it is not needed. I've run into this issue multiple times. I think it is good practice to wrap such things in a `section` or `namespace` to avoid making this mistake.

Reply:
I'm in favour of unifying global and finset disjoint. I don't see why `decidable_eq` would be an issue, since we're dealing with propositions, so there's no reason not to use `classical.dec_eq`?